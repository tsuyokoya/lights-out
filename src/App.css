.App {
  text-align: center;
}
body {
  background-color: black;
}

table,
td {
  border-collapse: collapse;
  border: 1px solid white;
}

h1 {
  font-family: Brush Script MT, Brush Script Std, cursive;
  font-size: 8em;
  margin: 0.25em 0;
  text-decoration: underline;
  color: #fff;
  animation: flicker 1.5s infinite alternate;
  animation-iteration-count: 5;
}

.win-msg {
  font-family: Brush Script MT, Brush Script Std, cursive;
  font-size: 8em;
  color: #fff;
}

@keyframes flicker {
  0%,
  13%,
  20%,
  25%,
  53%,
  57%,
  100% {
    text-shadow: 0 0 7px #fff, 0 0 10px #fff, 0 0 21px #fff, 0 0 42px #f09,
      0 0 82px #f09, 0 0 92px #f09;
  }
  20%,
  24%,
  55% {
    text-shadow: none;
  }
}
