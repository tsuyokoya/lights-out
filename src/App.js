import React from "react";
import Board from "./Board";
import "./App.css";

function App() {
  return (
    <div className="App">
      <h1>Lights Out</h1>
      <Board nrows={5} ncols={5} chanceLightStartsOn={0.5} />
    </div>
  );
}

export default App;
