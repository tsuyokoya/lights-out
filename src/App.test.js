import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

// smoke test
it("renders without crashing", () => {
  render(<App />);
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
