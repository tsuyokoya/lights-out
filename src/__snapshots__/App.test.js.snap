// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Lights Out
    </h1>
    <table
      class="Board"
    >
      <tbody>
        <tr>
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell "
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell "
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
          <td
            class="Cell Cell-lit"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell "
          />
          <td
            class="Cell "
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
        </tr>
        <tr>
          <td
            class="Cell "
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell Cell-lit"
          />
          <td
            class="Cell "
          />
          <td
            class="Cell Cell-lit"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
